code/datums/elements/backblast.dm

```diff
@@ -64,7 +66,12 @@
 /// For firing an actual backblast pellet
 /datum/element/backblast/proc/pew(turf/target_turf, obj/item/gun/weapon, mob/living/user)
 	//Shooting Code:
-	var/obj/projectile/bullet/incendiary/backblast/P = new (get_turf(user))
+	backblast_type = /obj/projectile/bullet/incendiary/backblast
+
+	if(locate(/obj/item/ammo_casing/caseless/rocket/solidfuel) in weapon)
+		backblast_type = /obj/projectile/bullet/incendiary/backblast/solidfuel
+
+	var/obj/projectile/bullet/incendiary/backblast/P = new backblast_type(get_turf(user))
```

Comment:
make this check if weapon is ballistic and then use the weapon's backblast type
something like:
```
var/obj/projectile/bullet/P
if (istype(weapon, /obj/item/gun/ballistic))
      var/obj/item/gun/ballistic/W = weapon
      P = new W.backblast(get_turf(user))
else
      P = new /obj/projectile/bullet/incendiary/backblast(get_turf(user))
```